### ISSUE TITLE
Generic.Formatting.SpaceBeforeCast.NoSpace sniff with parentheses

### ISSUE THREAD

**Issue by tominventisbe** [squizlabs/PHP_CodeSniffer#2469]
The rule `Generic.Formatting.SpaceBeforeCast.NoSpace` complains if there is no space before a cast, like so:

```php
return(string) $oops; // Wrong
return (string) $oops; // Okay
```

It however also complains when the cast is wrapped in parentheses:

```php
return ((string) $oops); // Wrong
```

I learned about this sniff via the release changelog and can't seem to find a list of sniffs and their explanations, so I couldn't really say, but is this intended behavior or a bug?

**Comment by jrfnl** [comment]
AFAICS this is intended behaviour. The sniff does not make allowances for the context in which a cast is used. It is basically intended for space-rich standards which always demand a space before a type cast.

The only "exception" is when the cast is preceded by a new line.

**Comment by gsherwood** [comment]
> AFAICS this is intended behaviour.

You wrote it, so I'd go with that :)

Personally, I'd allow no space if the previous token in an opening parentheses as this allows for fewer conflicts with other sniffs that ban spaces after an opening parentheses.
